manifests/init.pp

```diff
@@ -91,13 +91,19 @@
     },
   ])
 
+  exec { 'dry test servicenow.rb script':
+    command     => "${external_commands_base}/servicenow.rb __dry_test__",
+    subscribe   => $resource_dependencies,
+    refreshonly => true,
+  }
+
   ini_setting { 'puppetserver puppetconf trusted external command':
     ensure  => present,
     path    => '/etc/puppetlabs/puppet/puppet.conf',
     setting => 'trusted_external_command',
     value   => "${external_commands_base}/servicenow.rb",
     section => 'master',
     notify  => Service['pe-puppetserver'],
-    require => $resource_dependencies,
+    require => Exec['dry test servicenow.rb script'],
```

Comment:
Are we pretty sure that this isn't calling the exec with every agent run on the master?

Reply:
Yup, b/c of `refresh_only`.

Reply:
The idempotent test also confirms this.

Reply:
👍

Reply:
cool!